generator/generate.js

```diff
@@ -733,26 +731,39 @@ ${optPrefix("\n    // ", sanitizeComment(description))}
     }
   }
 
-  function printTsType(type, isRoot = true) {
+  function printTsType(field, name, canBeUndefined = true) {
+    let typeValue
+    let type
+
+    if (!name) {
+      name = field.name
+      type = field.type
+    } else {
+      type = field
```

Comment:
In the above three lines, can `field.type` and `field` be the same shape? i.e. can `field` and `field.type` have the same properties? Why do we need them both?

Reply:
yes, based on the way a field is described it could be single level or nested (in the case of being `NON_NULL`)

Reply:
👍